test/cmd/diagnostics.sh

```diff
@@ -0,0 +1,28 @@
+#!/bin/bash
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+OS_ROOT=$(dirname "${BASH_SOURCE}")/../..
+source "${OS_ROOT}/hack/util.sh"
+source "${OS_ROOT}/hack/cmd_util.sh"
+os::log::install_errexit
+
+# This test validates the diagnostics command
+
+# available diagnostics (2015-12-21):
+# AnalyzeLogs ClusterRegistry ClusterRoleBindings ClusterRoles ClusterRouter ConfigContexts DiagnosticPod MasterConfigCheck MasterNode NodeConfigCheck NodeDefinitions UnitStatus
+# Without things feeding into systemd, AnalyzeLogs and UnitStatus are irrelevant.
+# ConfigContexts runs into some invalid contexts created by other tests, so is omitted.
```

Comment:
> @deads2k @smarterclayton thoughts on these sorts of interactions between tests in test/cmd?
> 
> @sosiouxme I'd still recommend writing an end-to-end test instead, as it would allow you to run tests for your diagnostics that require nodes/pods running.

I don't think we need to worry about it too much for `test-cmd`.  `test-cmd` is about seeing if the basic structure of the command works.  It's not intended to be an exhaustive check of functionality.